examples/cloudml-churn-prediction/preprocessor/preprocessor/preprocess.py

```diff
@@ -0,0 +1,325 @@
+    else:
+        duration = datetime.now() - element['start_date']
+    duration_days = duration.days
+    element.update(duration=duration_days)
+    for index in range(0, len(classCeilings)):
+        if duration_days < classCeilings[index]:
+            element.update(label=features.LABEL_VALUES[index])
+            return element
+    element.update(label=features.LABEL_VALUES[len(classCeilings)-1])
+    return element
+
+
+def _combineCensorshipDuration(element):
+    """Transform users' duration and censorship indicator into a 2*n_intervals
+    array. For each row, the first half of the array indicates whether the user
+    survived the corresponding interval. Second half indicates whether the user
+    died in the corresponding interval. If the user is censored, the second
+    half is all zeros.
+
+    This function is required for both fake and real data.
+
+    Args:
+        element: Dictionary representing one sample
+    Returns:
+        element with following field added:
+            labelArray: 2D array where for each row (which corresponds to a 
+                user) first half of the values are 1 if the individual survived
+                that interval, 0 if not. The second half of the values are 1
+                for time interval during which failure occured (if uncensored)
+                and 0 for other intervals.
+    """
+    numIntervals = len(features.LABEL_CEILINGS)
+    y = np.zeros((numIntervals * 2))
+    breaks = [0] + features.LABEL_CEILINGS
+    duration = element['duration']
+    y[0:numIntervals] = 1.0*(np.full(numIntervals, duration) >= breaks[1:])
+    if not element['active']:
+        if duration < breaks[-1]:
+            y[numIntervals + np.where(np.full(numIntervals, duration) < breaks[1:])[0][0]] = 1.0
+    element.update(labelArray=y)
+    return element
+
+
+@beam.ptransform_fn
+def Shuffle(p):
+    """Shuffles the given pCollection."""
+    return (p
+            | "PairWithRandom" >> beam.Map(lambda x: (random.random(), x))
+            | "GroupByRandom" >> beam.GroupByKey()
+            | "DropRandom" >> beam.FlatMap(lambda x: x[1]))
+
+
+@beam.ptransform_fn
+def WriteTFRecord(p, prefix, output_dir, metadata):
+    """Shuffles and write the given pCollection as a TF-Record.
+    Args:
+        p: a pCollection.
+        prefix: prefix for location tf-record will be written to.
+        output_dir: the directory or bucket to write the json data.
+        metadata: 
+    """
+    coder = coders.ExampleProtoCoder(metadata.schema)
+    prefix = str(prefix).lower()
+    _ = (p
+        | "ShuffleData" >> Shuffle()  # pylint: disable=no-value-for-parameter
+        | "WriteTFRecord" >> beam.io.tfrecordio.WriteToTFRecord(
+            os.path.join(output_dir, 'data', prefix, prefix),
+            coder=coder,
+            file_name_suffix=".tfrecord"))
+
+
+@beam.ptransform_fn
+def randomly_split(p, train_size, validation_size, test_size):
```

Comment:
ptransform functions should follow the same naming convention (ie `RandomlySplit`)